### ISSUE TITLE
devops: Align Filesystem Hierarchy across CI, Docker and Rust code

### ISSUE THREAD

**Issue by gustavovalverde** [ZcashFoundation/zebra#7542]
### Describe the issue or request

Some documentation and Rust code, among other examples, uses `/zebrad-cache` as the path to save the cache state. We once decided to use the linux Filesystem Hierarchy Standard (FHS) for Docker, and changed some existing paths, including the one for cache state to `/var/cache/zebrad-cache`, this creates inconsistency and confusion for users and developers.

### Expected Behavior

Have the same convention across the whole project, including documentation.

### Current Behavior

Paths differ across the project

### Possible Solution

Use the Filesystem Hierarchy Standard (FHS) for all paths

### Additional Information/Context

I recall some paths were not possible to change (or at least not a good idea), as those were for matching `zcashd`

An user reported their cache state destroyed, based on this confusion: https://discord.com/channels/809218587167293450/809251029673312267/1150191217032826980

### Is this happening on PRs?

N/A

### Is this happening on the main branch?

N/A

**Comment by gustavovalverde** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Engineering-Team-5fb24d9264a3e8000e666a9e/issues/ZcashFoundation/zebra/7542?planning-poker) with Zenhub @arya2 @oxarbitrage @teor2345 @upbqdn

**Comment by mpguerra** [comment]
Is this related to #9456 ?

**Comment by gustavovalverde** [comment]
Yes, and to [main/docs/decisions/devops/0003-filesystem-hierarchy.md](https://github.com/ZcashFoundation/zebra/blob/main/docs/decisions/devops/0003-filesystem-hierarchy.md) also
